XenAdmin/ConsoleView/VNCTabView.cs

```diff
@@ -501,7 +508,17 @@ private void guestMetrics_PropertyChanged(object sender, PropertyChangedEventArg
         {
             if (e.PropertyName == "other")
             {
+                if (RDPEnabled)
+                {
+                    if (vncScreen.UseVNC && (tryToConnectRDP || (!vncScreen.UserWantsToSwitchProtocol && Properties.Settings.Default.AutoSwitchToRDP)))

```

Comment:
I Add some comments on code, and I can explain the logic here:
If current connection is VNC (which is "vncScreen.UseVNC == true" ), will switch to RDP connection as following conditions:
1. Click "Turn on Remote Desktop" which will set tryToConnectRDP = true
2. RDP status changed by turned on RDP in-guest and with "Automatically switch to the Remote Desktop console when it becomes available" is on (which is Properties.Settings.Default.AutoSwitchToRDP = true). But if user click button to chose connection type before (which set "vncScreen.UserWantsToSwitchProtocol" = true), we will not do the automatic switch.